### DISCUSSION TITLE
Single Transaction Migration Processing

### DISCUSSION THREAD

**Discussion by ryangurn** [laravel/framework#48190] [Ideas]
While using laravel I have noticed over time that when processing a batch of migrations, the ddl is being processed in individual transactions which leads to inconsistent rollbacks (if any). 

Would it make sense to process all of the migrations in a batch within a single transaction? It would ensure that if any of the ddl in one of the migrations causes a full rollback of entire batch. Does this already work this way in non-development `APP_ENV`?

Just a thought so that the database can stay more consistent if migrations result in exceptions.

**Comment by henzeb** [comment]
like with https://laravel.com/docs/10.x/migrations#squashing-migrations?

**Reply by ryangurn** [reply]
That would work since the entire script is run in one transaction. It just seems like it would be beneficial to run a batch of migrations in a single transaction.

I will take a look at squashing migrations though.
